src/client/dfs/dfs.c

```diff
@@ -347,15 +348,20 @@ fetch_entry(daos_handle_t oh, daos_handle_t th, const char *name,
 		if (rc) {
 			D_ERROR("Failed to fetch entry %s " DF_RC "\n", name,
 				DP_RC(rc));
+			D_FREE(value);
 			D_GOTO(out, rc = daos_der2errno(rc));
 		}
 
-		sym_len = sg_iovs[0].iov_len;
-		if (sym_len != 0) {
-			D_ASSERT(value);
-			D_STRNDUP(entry->value, value, PATH_MAX - 1);
-			if (entry->value == NULL)
-				D_GOTO(out, rc = ENOMEM);
+		entry->value_len = sg_iovs[0].iov_len;
+
+		if (entry->value_len != 0) {
+			/* Return value here, and allow the caller to truncate
+			 * the buffer if they want to
+			 */
+			entry->value = value;
```

Comment:
The large buffer is never returned from dfs, it's either freed in this file or the strdup is performed in this file.  The change is if the buffer is going to be freed anyway we never bother doing the additional allocation and copy.

Reply:
yea i was thinking in the case of a lookup which returns that to the use in the object. But I missed that you do the strndup there in that case, so that's resolves it.